cmd/http/server.go

```diff
@@ -129,6 +129,9 @@ func (srv *Server) Shutdown() error {
 	// Close underneath HTTP listener.
 	srv.listenerMutex.Lock()
 	err := srv.listener.Close()
+	if err != nil {
```

Comment:
It is just cleaner to unlock first and then return an error

Reply:
Yes agreed